[Linux] Enable frame pointers when building Swift libraries.

al45tair opened the pull request:
Turn on frame pointers for the Swift runtime libraries.  This makes backtraces that go through the runtimes more reliable without having to parse DWARF data.

rdar://116112040

al45tair committed:
[Linux] Enable frame pointers when building Swift libraries.

Turn on frame pointers for the Swift runtime libraries.  This makes
backtraces that go through the runtimes more reliable without having
to parse DWARF data.

rdar://116112040

al45tair commented:
@swift-ci Please test

al45tair commented:
**Explanation**: Set `-fno-omit-frame-pointer` when building the Swift runtime libraries.
**Original PR**: #68986
**Reviewed by**: TBD
**Resolves**: rdar://116112040
**Tests**: Adds a test that checks that we get a decent backtrace when someone calls `fatalError()`.

al45tair committed:
[Backtracing][Tests] Tweak the new fatalError test slightly.

The test needs tweaking a little to work on Linux and macOS.

rdar://116112040

al45tair commented:
@swift-ci Please test

shahmishal commented:
```
lit.py: /home/build-user/llvm-project/llvm/utils/lit/lit/TestingConfig.py:138: fatal: unable to parse config file '/home/build-user/swift/test/lit.cfg', traceback: Traceback (most recent call last):
  File "/home/build-user/llvm-project/llvm/utils/lit/lit/TestingConfig.py", line 127, in load_from_path
    exec(compile(data, path, 'exec'), cfg_globals, None)
  File "/home/build-user/swift/test/lit.cfg", line 585, in <module>
    config.substitutions.append( ('%swift-host-lib-dir', config.swift_host_lib_dir))
AttributeError: 'TestingConfig' object has no attribute 'swift_host_lib_dir'
```

al45tair commented:
That doesn't seem to be something I've changed.

al45tair commented:
Yeah, that's #68858. It deleted
```
config.swift_host_lib_dir = make_path(config.swift_lib_dir, 'swift', 'host')
```
but didn't remove
```
config.substitutions.append( ('%swift-host-lib-dir', config.swift_host_lib_dir))
```

al45tair committed:
[Test] Fully remove config.swift_host_lib_dir.

Somehow #68858 managed to remove `config.swift_host_lib_dir` without
removing the corresponding `config.substitutions.append`.  Fix.

rdar://116112040

al45tair commented:
@swift-ci Please test
